### PR TITLE
Use env var to explicitly specify rdkafka setup

### LINKED ISSUE #920: librdkafka dependency and setup issues
**PyKafka version**: 2.8.0

In setup.py, with CPython, it detects the existence of librdkafka and compile the extension by default. When it fails, it silently disables the librdkafka support.

When I set up the environment for librdkafka, I installed the system `librdkafka-dev` then I ran setup.py. And I actually bump into a silent error when I ran `pip install -e .`. I had to run `setup.py install` to see the error message:
```
running build_ext
building 'pykafka.rdkafka._rd_kafka' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/include/python2.7 -c pykafka/rdkafka/_rd_kafkamodule.c -o build/temp.linux-x86_64-2.7/pykafka/rdkafka/_rd_kafkamodule.o
pykafka/rdkafka/_rd_kafkamodule.c: In function 'Producer_delivery_report_callback':
pykafka/rdkafka/_rd_kafkamodule.c:568:9: error: unknown type name 'rd_kafka_timestamp_type_t'
        rd_kafka_timestamp_type_t timestamp_type;
        ^
pykafka/rdkafka/_rd_kafkamodule.c:569:29: warning: implicit declaration of function 'rd_kafka_message_timestamp' [-Wimplicit-function-declaration]
        int64_t timestamp = rd_kafka_message_timestamp(rkmessage, &timestamp_ty
                            ^
pykafka/rdkafka/_rd_kafkamodule.c:570:31: error: 'RD_KAFKA_TIMESTAMP_NOT_AVAILABLE' undeclared (first use in this function)
        if (timestamp_type != RD_KAFKA_TIMESTAMP_NOT_AVAILABLE) {
                              ^
pykafka/rdkafka/_rd_kafkamodule.c:570:31: note: each undeclared identifier is reported only once for each function it appears in
pykafka/rdkafka/_rd_kafkamodule.c: In function 'Consumer_consume':
pykafka/rdkafka/_rd_kafkamodule.c:978:9: error: unknown type name 'rd_kafka_timestamp_type_t'
        rd_kafka_timestamp_type_t timestamp_type;
        ^
pykafka/rdkafka/_rd_kafkamodule.c:980:31: error: 'RD_KAFKA_TIMESTAMP_NOT_AVAILABLE' undeclared (first use in this function)
        if (timestamp_type == RD_KAFKA_TIMESTAMP_NOT_AVAILABLE) {
```

1. Is it possible to let users specify rdkafka support and fail the setup if the compilation fails? If the user expects rdkafa support, failing at setup sounds better than failing with ImportError at runtime.
2. Should we document the version of librdkafka needed for the extension to compile successfully? I got librdkafka 0.8.6 (the latest in apt for Ubuntu 16.04) but it doesn't compile. When I manually compile the v0.11.6 from GitHub, it works. There's a similar issue in confluent-kafka-python https://github.com/confluentinc/confluent-kafka-python/issues/230
3. Should we vendor librdkafka? git clone, git checkout to tag, compile and use that shared library file (or use extension)?

I could help with the fix and the doc but there are some decisions to be made. Thanks!

### PR DESCRIPTION
**Pull request by carsonip**
Read RDKAFKA_INSTALL env var on setup.
When RDKAFKA_INSTALL='system', compile rdkafka c ext. Fail setup on
compilation error.
When RDKAFKA_INSTALL='', don't compile rdkafka c ext.

No change on default behavior when RDKAFKA_INSTALL is not specified /
invalid value.

Fix #920

### CODE REVIEW

**Comment by carsonip** [comment]
By not using a boolean-style env var, we may accept new values in the future e.g. 'vendor' that will download, and compile rdkafka for users.

File: setup.py [hunk 1]
```diff
@@ -184,17 +184,30 @@ def run_setup(with_rdkafka=True):
         ]
     )
 
+
+# Use environment variable RDKAFKA_INSTALL to explicitly specify whether rdkafka c extension should be compiled
+# 'system': Compile with librdkafka installed in system
+# '' (empty string): No compile
+install_type = os.environ.get('RDKAFKA_INSTALL')
```
[the remark this answers is not in the document]
Comment (carsonip) [reply setup.py]: My idea was if RDKAFKA_INSTALL env var is not specified (i.e. `install_type` is None), do the default action = try compiling rdkafka c ext if cpython. This make things backward-compatible.

If the user does `RDKAFKA_INSTALL=''`, which is specifying the env var and assigning it to empty string, this means the c ext should not be compiled. This is more like `RDKAFKA_INSTALL=no`, but in a way that no one has to remember whether this env var should be set to 0, 'no' or 'false'. I'm not sure what's the common practice out there.

[the remark this answers is not in the document]
Comment (carsonip) [reply setup.py]: I'll deal with the naming / formatting issues once we finish our discussion on the env var.

[the remark this answers is not in the document]
Comment (emmett9001) [reply setup.py]: Oh, that's an important distinction. Thanks for the explanation.

**Comment by carsonip** [comment]
Pushed a commit for documentation. This PR should be ready. Please squash merge if it looks good to you. Thanks @emmett9001

**Pull request merged**